control/control_proxy.go

```diff
@@ -0,0 +1,237 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package control
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/intelsdi-x/snap/control/rpc"
+	"github.com/intelsdi-x/snap/core"
+	"github.com/prometheus/common/log"
+	"golang.org/x/net/context"
+)
+
+type controlProxy struct {
+	control *pluginControl
+}
+
+func (pc *controlProxy) Load(ctx context.Context, arg *rpc.PluginRequest) (*rpc.PluginReply, error) {
```

Comment:
This load method is incompatible with auto discovery. Auto discovery passes a path as we do not write the plugin to a temporary directory. This implementation is fine for plugins loaded from a REST API, but just gives more weight behind auto discovery needing to be moved inside of control.